storm-server/src/main/java/org/apache/storm/scheduler/resource/ResourceAwareScheduler.java

```diff
@@ -202,7 +210,7 @@ private void scheduleTopology(TopologyDetails td, Cluster cluster, final User to
                             if (evictAssignemnt != null && !evictAssignemnt.getSlots().isEmpty()) {
                                 Collection<WorkerSlot> workersToEvict = workingState.getUsedSlotsByTopologyId(topologyEvict.getId());
                                 topologySchedulingResources.adjustResourcesForEvictedTopology(toSchedule, topologyEvict);
-
+                                evictedTopologies.add(topologyEvict.getId());
```

Comment:
One last thing, we can have a boolean like `forTests`, which by default to false and we can set it to true from unit tests. This is to avoid unnecessary computation/operations. And make it clear that this is only for tests. 

As for `getEvictedTopologies`, we can throw exceptions if `forTests` is false.

Reply:
Addressed.